chrome-extension/sidebar/sidebar.js

```diff
@@ -48,6 +48,307 @@ const htmlToElement = (html) => {
   return template.content.firstChild;
 };
 
+// Begin code for sorting course list
+document.addEventListener('change', function(event) {
```

Comment:
This pattern is bad, we should add a listener to the target instead of listening to the whole document. Listening to the whole document leads to a lot of excessive computation